### ISSUE TITLE
Build failure for v0.88.0

### ISSUE THREAD

**Issue by p-linnane** [anchore/syft#2066]
**What happened**:
```
go: updates to go.mod needed; to update it:
	go mod tidy
```

**What you expected to happen**:
syft builds

**Steps to reproduce the issue**:
Build syft from source

**Anything else we need to know?**:
Relates to https://github.com/Homebrew/homebrew-core/pull/140496

**Comment by blacktop** [comment]
Latest is broken for me too

**Comment by kzantow** [comment]
Hi @p-linnane and @blacktop -- are you both using **go 1.21** to compile? Using 1.19, I'm not seeing any problems like this at the moment.

**Comment by kzantow** [comment]
@blacktop is your goreleaser pipeline something that you could share?

**Comment by sbrun** [comment]
I have this issue too
It fails to build from source when using go 1.21 with the error message
```
go: updates to go.mod needed; to update it:
        go mod tidy
```
It doesn't fail with go 1.20.

**Comment by kzantow** [comment]
Thanks @sbrun; I'm still waiting to hear if @p-linnane is also using Go 1.21, but I'm going to assume so.

Let me explain the issue: in Go 1.21, a new keyword: `toolchain` was introduced as a part of the `go.mod`. Since this keyword is not present in the [Syft go.mod](https://github.com/anchore/syft/blob/main/go.mod#L3), Go 1.21 complains the go.mod should be updated (as you see with the error _"go: updates to go.mod needed; to update it: go mod tidy"_). HOWEVER, Go 1.19 and 1.20 do not have support for this keyword, which means if we update the go.mod to be compatible with 1.21, we _break compatibility with older versions_, which we do not want to do at this time.

That said, it appears the Go team is backporting support for the `toolchain` keyword, [for example the go 1.19 issue here](https://github.com/golang/go/issues/61197). But this has [not been released in a go 1.19 patch release yet](https://go.dev/doc/devel/release#go1.19.minor), which leaves us in the situation that we cannot update the go.mod in a way that is compatible with both 1.19 AND 1.21 _until_ a patch release of 1.19 is released. As of today, the latest version of 1.19 is go1.19.12, released Aug 1. We expect early next month to get a patch release that presumably includes support for the `toolchain` keyword, given that the issue is closed as completed. Once a 1.19 patch is released with this backport, we should be able to update the go.mod to remain on Go 1.19 and also have the `toolchain` keyword included, which _should_ support compiling on 1.21 without the aforementioned error.

TL;DR: we aren't going to break go 1.19 compatibility, but we believe if we just _wait_ a short time for an updated go 1.19, we should be able to solve this issue.

**Comment by sbrun** [comment]
Thanks for the detailed explanations.

**Comment by p-linnane** [comment]
Yes, we are packaging syft for Homebrew. Homebrew always uses the latest versions of dependencies.

**Comment by kzantow** [comment]
@p-linnane apologies for being relatively unfamiliar with the Homebrew processes, but is there a way to pin the Syft build to go 1.19?

**Comment by p-linnane** [comment]
We can, but it's not ideal. Only 23 of the 855 packages that depend on Go in Homebrew are on versions of Go prior to 1.21.

We've incorporated the changes in https://github.com/anchore/syft/pull/2067 to fix the Go 1.21 build.

**Comment by wagoodman** [comment]
Point taken on 1.19 being EOL. One clarification, the `go` keyword is the minimum supported version of go for the module, not necessarily the version of go that must be used by the end user application. Using go 1.20 would be the next best version to use in this case until the upcoming go 1.19/1.20 release in hopefully a week or so.

**Comment by chenrui333** [comment]
I can confirm it is working fine now. Thanks!

**Comment by kzantow** [comment]
Thanks @chenrui333 -- I'm going to close this issue since we've moved to Go 1.21.
